aospy/utils/times.py

```diff
@@ -562,13 +562,16 @@ def ensure_time_as_index(ds):
     Dataset
 
     """
-    time_indexed_vars = set(ds.data_vars).union({TIME_WEIGHTS_STR,
-                                                 TIME_BOUNDS_STR})
+    time_indexed_coords = {TIME_WEIGHTS_STR, TIME_BOUNDS_STR}
+    time_indexed_vars = set(ds.data_vars).union(time_indexed_coords)
     time_indexed_vars = time_indexed_vars.intersection(ds.variables)
     for name in time_indexed_vars:
         if TIME_STR not in ds[name].indexes:
             da = ds[name].expand_dims(TIME_STR)
             da[TIME_STR] = ds[TIME_STR]
             ds[name] = da
     # Convert time weights and bounds back to coordinates from data_vars
-    return ds.set_coords([TIME_WEIGHTS_STR, TIME_BOUNDS_STR])
+    for coord in time_indexed_coords:
+        if coord in ds.coords:
```

Comment:
Should this be `if coord not in ds.coords`?  Otherwise this loop is a no-op right?  

More generally though, I'm not sure if this logic is needed, because we call `set_grid_attrs_as_coords` directly after `_prep_time_data`, and all the functions called within `_prep_time_data` operate on Datasets:
https://github.com/spencerahill/aospy/blob/7e3e5becb6920c3c5987a6dfd0a398c85d918ae4/aospy/data_loader.py#L278-L279

Reply:
> Should this be `if coord not in ds.coords`?  Otherwise this loop is a no-op right?  

To cover the case where `TIME_WEIGHTS_STR` or `TIME_BOUNDS_STR` are not in the Dataset, I mean:  `if coord in ds.data_vars`

Reply:
Good call.  For some reason while putting the tests together, I convinced myself that time_bounds and time_weights were being promoted to data_vars.  But the full suite passed on my machine with this omitted, so clearly that was mistaken.